src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -1375,4 +1394,119 @@ private ITextComponent onGetDisplayName4(IMerchant villager) {
         }
         return new TextComponentString(SpongeTexts.toLegacy(this.displayName));
     }
+
+    @Override
+    @Nullable
+    public ProfileProperty getSkin() {
+        return Iterables.getFirst(this.getProfile().getPropertyMap().get(ProfileProperty.TEXTURES), null);
+    }
+
+    @Override
+    public boolean setSkin(ProfileProperty skin) {
+        Collection<ProfileProperty> props = this.getProfile().getPropertyMap().get(ProfileProperty.TEXTURES);
+        props.clear();
+        props.add(skin);
+
+        this.updateSkin();
+        return true;
+    }
+
+    @Override
+    public DataTransactionResult removeSkin() {
+        if (!this.getProfile().getPropertyMap().containsKey(ProfileProperty.TEXTURES)) {
+            return DataTransactionResult.successNoData();
+        }
+        Collection<ProfileProperty> skin = this.getProfile().getPropertyMap().removeAll("textures");
+        ImmutableValue<?> oldValue = new ImmutableSpongeValue<>(Keys.SKIN, skin.iterator().next());
+        this.updateSkin();
+
+        return DataTransactionResult.builder().result(DataTransactionResult.Type.SUCCESS).replace(oldValue).build();
+    }
+
+    private void updateSkin() {
+        this.updateSkinSelf();
+        this.updateSkinOthers();
+    }
+
+    private void updateSkinSelf() {
+        this.getTabList().removeEntry(this.getUniqueId());
+        this.getTabList().addEntry(TabListEntry.builder()
+                .displayName(this.getDisplayNameData().displayName().get())
+                .latency(this.getConnection().getLatency())
+                .list(this.getTabList())
+                .gameMode(this.getGameModeData().type().get())
+                .profile(this.getProfile())
+                .build());
+
+        if (this.isDead) {
+            return;
+        }
+
+        // Simulate respawn to see skin active
+        this.fakeRespawn();
+        /*Location<World> loc = this.getLocation();
+        Vector3d rotation = this.getRotation();
+
+        WorldProperties other = null;
+        for (WorldProperties w : Sponge.getServer().getAllWorldProperties()) {
+            if (other != null) {
+                break;
+            }
+
+            if (!w.getUniqueId().equals(this.getWorld().getUniqueId())) {
+                Sponge.getServer().loadWorld(w.getUniqueId());
+                other = w;
+            }
+        }
+
+        if (other != null) {
+            this.setLocation(Sponge.getServer().getWorld(other.getUniqueId()).get().getSpawnLocation());
+            this.setLocationAndRotation(loc, rotation);
+        }*/
+    }
+
+    // Adapted from MixinPlayerList#recreatePlayerEntity
+    // We want to respawn the player *only* on tts own client. Therefore,
+    // we skip most of the logic that's performed for a normal respawn.
+    private void fakeRespawn() {
```

Comment:
This method is pretty hacky, but I don't believe it should cause any problems. The respawning all happens within one server tick, and should be completely transparent to plugins.